### PR TITLE
Added middlewares; fixes; added makefile demo command.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/loadbalancer/main.go [hunk 1]
```diff
@@ -47,8 +47,8 @@ func main() {
 
 	// Register standard Go metric collectors, which are by default registered when using global registry.
 	reg.MustRegister(
-		version.NewCollector("observable-demo"),
-		prometheus.NewBuildInfoCollector(),
+		version.NewCollector(""),
+		prometheus.NewBuildInfoCollector(), // Why both? Can we choose one?
```
Comment (bwplotka) [inline cmd/loadbalancer/main.go]: ping @kakkoyun

File: cmd/loadbalancer/main.go [hunk 2]
```diff
@@ -47,8 +47,8 @@ func main() {
 
 	// Register standard Go metric collectors, which are by default registered when using global registry.
 	reg.MustRegister(
-		version.NewCollector("observable-demo"),
-		prometheus.NewBuildInfoCollector(),
+		version.NewCollector(""),
```
Comment (bwplotka) [inline cmd/loadbalancer/main.go:50]: I like the idea of doing `build_info` only so you can have nice version dashboard finally (:

Or use just `NewBuildInfoCollector` for it (:

File: pkg/exthttp/middleware.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
+package exthttp
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type ServerMetrics struct {
```
Comment (bwplotka) [inline pkg/exthttp/middleware.go:10]: This is bit changed version:
* No interface, no nop (why it was needed even?)
* InFlight gauge
* handler reuses Wrap technique
* No method just function for new handler, no need to combine those together IMO.

File: cmd/loadbalancer/main.go (hunk 1 above)
Comment (kakkoyun) [reply cmd/loadbalancer/main.go]: They include different extra information.
Version:

https://github.com/prometheus/common/blob/629b6ff9b3aafafba54ab96663903fde9544f037/version/info.go#L27-L34

Build Info:
https://github.com/prometheus/client_golang/blob/803ef2a759d7caaaa0de58e3815f1be4c8b5a42a/prometheus/go_collector.go#L388-L392

When I'm in doubt I choose both :D

If I had to choose one, I'd choose `version.NewCollector`

File: cmd/loadbalancer/main.go [hunk 4]
```diff
@@ -68,34 +66,85 @@ func main() {
...
+			defer cancel()
+
+			if err := srv.Shutdown(ctx); err != nil {
+				log.Println("error: server shutdown failed")
+			}
+		})
+	}
+	{
+		srv := &http.Server{
+			// This ensures no keep-alive and thus always startin new connection for demo purposes.
```
Comment (kakkoyun) [inline cmd/loadbalancer/main.go]:
```suggestion
			// This ensures no keep-alive and thus always starting new connections for demo purposes.
```

**Review by kakkoyun — APPROVED** [review]
I love the improvements.
Simpler and more clear.

**Comment by bwplotka** [comment]
lint is super strict, I would lose it a bit ;p

**Pull request merged**